python/docs/source/user_guide/sql/python_data_source.rst

```diff
@@ -59,8 +59,17 @@ Start by creating a new subclass of :class:`DataSource`. Define the source name,
         def reader(self, schema: StructType):
             return FakeDataSourceReader(schema, self.options)
 
+        def streamReader(self, schema: StructType):
+            return FakeStreamReader(schema, self.options)
 
-**Step 2: Implement the Reader**
+        def simpleStreamReader(self, schema: StructType):
+            return SimpleStreamReader()
+
+        def streamWriter(self, schema: StructType, overwrite: bool):
+            return FakeStreamWriter(self.options)
+
```

Comment:
Actually, shall we add a new section for streaming data sources instead of mixing it with the batch one?

Reply:
Section separated.